DuckDuckGo/AppDelegate.swift

```diff
@@ -57,22 +58,33 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
 
     func applicationDidBecomeActive(_ application: UIApplication) {
         guard !testing else { return }
+        
         Pixel.fire(pixel: .appLaunch)
         StatisticsLoader.shared.load()
         startOnboardingFlowIfNotSeenBefore()
+        
         if appIsLaunching {
             appIsLaunching = false
-            autoClear = AutoClear(worker: mainViewController!)
-            autoClear?.applicationDidLaunch()
-            AppConfigurationFetch().start(completion: nil)
+            onApplicationLaunch(application)
+        }
+    }
+    
+    private func onApplicationLaunch(_ application: UIApplication) {
+        if privacyStore.authenticationEnabled {
             displayAuthenticationWindow()
             beginAuthentication()
-            initialiseBackgroundFetch(application)
         }
+        
+        autoClear = AutoClear(worker: mainViewController!)
+        autoClear?.applicationDidLaunch()
+        AppConfigurationFetch().start(completion: nil)
+        initialiseBackgroundFetch(application)
     }
 
     func applicationWillEnterForeground(_ application: UIApplication) {
-        beginAuthentication()
+        if privacyStore.authenticationEnabled {
+            beginAuthentication()
+        }
```

Comment:
`beginAuthentication()` has a side-effect: it always calls `removeOverlay()`. :)

As a result, this scenario fails:
1. Enable auto clear feature. Make sure App Lock is disabled.
2. Background the app.
3. Move back to the app - blank screen overlay is not removed.

Just calling `removeOverlay()` in `else` should fix that (unless you have some other small scale refactoring in mind that makes more sense to you :) ).

Reply:
Oh nice catch, that's fixed.

Reply:
@subsymbolic I'm afraid you've added that `else` in the wrong place: it should be here (in `applicationWillEnterForeground`) and it is not needed in `onApplicationLaunch` as there is no overlay on app startup.

Reply:
Oh dear, I should try not to code when I'm talking to do maths over chat. Thanks @bwaresiak, will fix.

Reply:
Now it works as expected. 👍